lib/core/task.toit

```diff
@@ -170,57 +172,53 @@ class Task_ implements Task:
       exit_ exception != null
 
   exit_ has_error/bool:
-    if has_error: exit_with_error_ = true
+    if has_error: task_exit_with_error_ = true
     task_count_--
     if background: task_background_--
     // Yield to the next task.
     next := suspend_
     if timer_:
       timer_.close
       timer_ = null
-    task_transfer_ next true  // Passing null will detach the calling execution stack from the task.
+    task_transfer_to_ next true
 
   suspend_:
-    previous := previous_running_
-    if previous == this:
-      // If we encounted a root-error, terminate the process.
-      if exit_with_error_: __exit__ 1
-      // Check whether no service definitions and only background tasks are running.
-      if ServiceManager_.is_empty and task_count_ == task_background_: __halt__
-      is_task_idle_ = true
+    processed := false
+    state_ = STATE_SUSPENDING
+    if identical this previous_running_:
```

Comment:
Is it necessary to make this 'if'?
Is it, so we give every task a chance to run when one task exited with an error? If yes, add a comment.

Reply:
Fixed the behavior to eagerly exit.